sparse/utils.py

```diff
@@ -240,3 +242,44 @@ def normalize_axis(axis, ndim):
         return tuple(normalize_axis(a, ndim) for a in axis)
 
     raise ValueError("axis %s not understood" % axis)
+
+
+def equivalent(x, y):
+    return (x == y) | ((x != x) & (y != y))
```

Comment:
note: if you're calling this on arrays, you could skip the non-NA checks for dtypes that can't hold NaN/NaT.